lib/EnsEMBL/REST/Model/ga4gh/variants.pm

```diff
@@ -376,7 +382,41 @@ sub getSingleCallSets{
 
 }
 
+## nameless (fake?) variants not in the database
+## id is based on chrom and location
+## create private GA4GH database?
+sub get_compliance_variant{
+
+  my ($self, $variantSetId, $varname ) = @_;
+
+  return  unless $variantSetId ==11;
+
+  ## create post style input structure
+  my $data;
+  my ($chr, $pos) = split/\_/, $varname ;
+
+  $data->{start}         = $pos - 1;
+  $data->{end }          = $pos;
+  $data->{referenceName} = $chr;
+  $data->{variantSetId}  = $variantSetId; 
 
+  ## load VCFcollections object for variantSet 
+  $data->{vcf_collection} = $self->context->model('ga4gh::ga4gh_utils')->fetch_VCFcollection_by_id($data->{variantSetId});
+  $self->context()->go( 'ReturnError', 'custom', [ " Failed to find the specified variantSetId"])
```

Comment:
Don't do the controller's job in the model. Models are not supposed to trigger responses to users, they should only throw exceptions for the controller to handle, or else return some data.